[CI] LangMustacheClientYamlTestSuiteIT.test {yaml=lang_mustache/30_search_template/Test with new response format} failed

`LangMustacheClientYamlTestSuiteIT.test {yaml=lang_mustache/30_search_template/Test with new response format}` failed on my PR build and seems unrelated:

https://scans.gradle.com/s/axrx54rul6s3c/tests/wvxejxkegxw3s-v3qw2yd63iq4w

```
java.lang.AssertionError: 
Failure at [lang_mustache/30_search_template:148]: hits.total.value didn't match expected value:
              hits.total.value: expected [0] but was [1]
```

Did not reproduce locally with the reproduction line.

--- a/modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/30_search_template.yml
+++ b/modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/30_search_template.yml
@@ -128,16 +128,12 @@
 
 ---
 "Test with new response format":
-
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/42664"
-
   - do:
       index:
         index:  test
         id:     1
         body:   {}
+        refresh: true
 
   - do:
       put_script:
@@ -150,5 +146,5 @@
       search_template:
         body: { "id": "template_1", "params": {} }
 
-  - match: { hits.total.value: 0 }
+  - match: { hits.total.value: 1 }
   - match: { hits.total.relation: eq }

